nessie/src/main/java/org/apache/iceberg/nessie/NessieTableOperations.java

```diff
@@ -174,6 +185,38 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
     }
   }
 
+  private static void handleReferenceConflictException(NessieReferenceConflictException ex) {
+    // Check if the server returned 'ReferenceConflicts' information
+    ReferenceConflicts referenceConflicts = ex.getErrorDetails();
+    if (referenceConflicts == null) {
+      return;
+    }
+
+    // Can only narrow down to a single exception, if there is only one conflict.
+    List<Conflict> conflicts = referenceConflicts.conflicts();
+    if (conflicts.size() != 1) {
+      return;
+    }
+
+    Conflict conflict = conflicts.get(0);
+    ConflictType conflictType = conflict.conflictType();
+    if (conflictType != null) {
+      switch (conflictType) {
+        case NAMESPACE_ABSENT:
+          throw new NoSuchNamespaceException(ex, "Cannot find namespace %s", conflict.key());
```

Comment:
would it make more sense to slightly adjust this to `"Namespace does not exist: %s"`?

Reply:
Not sure why the message matters.

Reply:
this is to align with other error msg patterns across the codebase